sql/table_cache.cc

```diff
@@ -594,8 +592,11 @@ bool tdc_init(void)
   mysql_cond_register("sql", all_tc_conds, array_elements(all_tc_conds));
 #endif
   /* Extra instance is allocated to avoid false sharing */
-  if (!(tc= new Table_cache_instance[tc_instances + 1]))
+  size_t size= sizeof(Table_cache_instance) * (tc_instances + 1);
+  if (!(tc_mem = static_cast<Table_cache_instance*>(
+                 aligned_malloc(size, CPU_LEVEL1_DCACHE_LINESIZE))))
```

Comment:
Do we really need any `static_cast` here? `tc_mem` expects `void*`, which is the return type of `aligned_malloc()`.

Reply:
I think that we will also need
```c++
static_assert(!((sizeof *tc) % CPU_LEVEL1_DCACHE_LINESIZE), "alignment");
```
because that is what we should be assuming here, to have each element of the array aligned. It may be necessary to retain a `pad[]` with a reasonable size, like we do in `btr_search_sys_t::partition::pad`.